src/app/buyeroffer/updateOffer.component.ts

```diff
@@ -34,23 +40,62 @@ export class UpdateOfferComponent implements OnInit {
       );
   }
 
+  toggleDeleteAdvertisementConfirm() {
+    this.hasDeleteConfirm = true;
+    setTimeout(function() {
+      this.hasDeleteConfirm = false;
+      console.log(this.edited);
+    }.bind(this), 3000);
```

Comment:
In TypeScript and ES6 you can use arrow functions instead, which bind on lexical scope, resulting in:
`() => { this.hasDeleteConfirm = false; }`